pkg/settings/setting.go

```diff
@@ -108,7 +108,11 @@ func PrefixPrivateRegistry(image string) string {
 }
 
 func IsRelease() bool {
-	return releasePattern.MatchString(ServerVersion.Get())
+	return IsReleaseHelper(ServerVersion.Get())
+}
+
+func IsReleaseHelper(input string) bool {
+	return !strings.Contains(input, "head") && releasePattern.MatchString(input)
```

Comment:
Can we just avoid the new helper function? I see that you need the helper function to define a unit test for this, but since the logic here is a simple string comparision which might be entirely [replaced with `isValid` based on your comment](https://github.com/rancher/rancher/pull/29525#issuecomment-709437839) I'm not sure if we need the unit test.

Without the helper function, this just becomes a 2 line change.

```suggestion
        server := ServerVersion.Get()
        return !strings.Contains(server, "head") && releasePattern.MatchString(server)
}
```

Reply:
cc: @ibuildthecloud if you had a specific reason for recommending it?

Reply:
Using ```semver.IsValid``` would have been my first choice, but I believe the goal was to avoid unintended collateral damage. Therefore, just checking for the ```head``` substring was enough.

As far as the unit test goes, alternatively, we could use ```IsRelease(optionalInput ...string) bool``` without creating the helper function that the test relies on. However, I wanted to avoid changing the original implementation since ```ServerVesion.Get()``` is hardcoded. There are multiple ways to tackle this, so long as we think a unit test is necessary.

Reply:
Agree that we should avoid helper function just for unit test. I would try to mock the setting. If that doesn't work then this is basic enough of a change that we don't need a unit test.

Reply:
Keep the test, drop the helper and set your server version in the tests. And thank you for adding tests

Reply:
@dramich @cbron acknowledged, and will move accordingly.

Reply:
Just published the new PR. @aiyengar2 kept your recommendations but did not want to introduce another state, so I called the function twice instead (simple getter method).